app/models/user.rb

```diff
@@ -366,4 +382,18 @@ def self.seed_data
     File.exist?(seed_file_name) ? YAML.load_file(seed_file_name) : []
   end
   private_class_method :seed_data
+
+  private
+
+  def unlock_queue
+    queue_opts = {
+      :class_name  => self.class.name,
+      :instance_id => id,
+      :method_name => 'unlock!',
+      :deliver_on  => Time.now.utc + ::Settings.authentication.locked_account_timeout,
+      :priority    => MiqQueue::MAX_PRIORITY
+    }
+
+    MiqQueue.put_or_update(queue_opts)
```

Comment:
@Fryguy somehow this never updates an existing item in the queue, I guess I'm doing something wrong with the arguments. I don't think it's okay that we queue the task N times if the user fails N+3 times.

Reply:
On the other hand, if the account is locked, this method will be never called. So this can be only called explicitly from the console. Anyway, it's still strange how the `put_or_update` is behaving.

Reply:
It's probably not updating because the deliver_on changes, and I think all parameters are checked for match.  There is a way to use a subset of parameter for check vs update.

Reply:
I think you need to do:

```ruby
    MiqQueue.put_or_update(
      :class_name  => self.class.name,
      :instance_id => id,
      :method_name => 'unlock!',
      :priority    => MiqQueue::MAX_PRIORITY
    ) do |_msg, queue_options|
      queue_options.merge(:deliver_on => Time.now.utc + ::Settings.authentication.locked_account_timeout.to_i)
    end
```

Reply:
It works, thanks!